### PR TITLE
Update AutoFuel.lua

### PR DESCRIPTION
**Pull request by RodrickLord**
Updating accoring to idea in TODO (slot.fuel)
Also, the turtle will automatically find a fuel

### CODE REVIEW

File: src/Turtle/AutoFuel.lua [hunk 1]
```diff
@@ -1,13 +1,29 @@
 --Author: Lupus590
+--Modified by ItsRodrick / RodrickLord
+-- (I saw slot.fuel in the TODO list)
 
 if(not turtle) then error("script must be run on a turtle",2) end--the 2 tell lua to blame the line that called this script
 if(turtle.getFuelLevel() == "unlimited") then return end--script is useless with unlimited fuel, so exit
 --print("Warning: automatic refuel is enabled, items may disappear if the turtle runs low on fuel.\n".."To prevent this make sure that the turtles fuel level stays above "..toString(fuelReserveLevel).."\n".."the automatic refuel script will take fuel one at a time until the above value is reached or exceeded.") 
+local function findFuel()
+	slot = slot or {} --If slot is nil, set it to a table
```
Comment (KingofGamesYami) [inline src/Turtle/AutoFuel.lua]: Why?  It makes no sense to make it a table...

Comment (lupus590) [inline src/Turtle/AutoFuel.lua]: Agreed.

On a separate note, this code may "interrupt" another program, please put the selected slot back to the one it was before the turtle refuelled.

File: src/Turtle/AutoFuel.lua [hunk 2]
```diff
@@ -1,13 +1,29 @@
...
+			slot.fuel = i
+			return i
+		end
+	end
+end
+
 local function autoFuelBackground()
 	local fuelReserveLevel = 10
 	
+	if not slot.fuel then findFuel() end
```
Comment (KingofGamesYami) [inline src/Turtle/AutoFuel.lua]: You may want to add a check inside the while loop, in case the current fuel slot runs out (or is moved by a program).

**Comment by KingofGamesYami** [comment]
Because of issues with your code, I doubt it will be used.  You may correct your mistakes and send another pull request, but until then I am closing this one.

**Pull request closed without merging**